pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -550,7 +552,10 @@ protected void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
                         readType == ReadType.Replay);
 
                 c.sendMessages(entriesForThisConsumer, batchSizes, batchIndexesAcks, sendMessageInfo.getTotalMessages(),
-                        sendMessageInfo.getTotalBytes(), sendMessageInfo.getTotalChunkedMessages(), redeliveryTracker);
+                        sendMessageInfo.getTotalBytes(), sendMessageInfo.getTotalChunkedMessages(), redeliveryTracker)
+                        .addListener(future -> {
+                    topic.recordReadLatency(System.nanoTime() - startTimeNs, TimeUnit.NANOSECONDS);
```

Comment:
The `startTimeNs` might be changed during the dispatch process, so that we will get a inaccurate dispatch latency

Reply:
OK  I will fix